pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTAnyTypeBodyDeclaration.java

```diff
@@ -15,48 +15,12 @@
     /**
      * Returns the child of this declaration,
      * which can be cast to a more specific node
-     * type using {@link #getKind()} as a cue.
+     * type using #getKind() as a cue.
```

Comment:
I think, there is no `#getKind()` method anymore, is it? The approach would now be testing with instanceof. Or could we return a more specific type than JavaNode now?

Reply:
In fact I have a pr lined up to make an interface for those nodes. This would remove ASTAnyTypeBodyDeclaration and subclasses, which are useless now. So eg a MethodDeclaration, FieldDeclaration would be direct children of the body